python/start-cluster.py

```diff
@@ -0,0 +1,212 @@
+    parser.add_argument(
+        "-y",
+        "--yaml",
+        type=str,
+        required=True,
+        help="Path to the YAML configuration file",
+    )
+    parser.add_argument(
+        "-l",
+        "--logfile",
+        type=str,
+        required=False,
+        default="neuraldb_enterprise.log",
+        help="Path to the log file",
+    )
+
+    return parser.parse_args()
+
+
+def main():
+    args = parse_arguments()
+
+    logger = LoggerConfig(log_file=args.logfile).get_logger(args.logfile)
+
+    user_config = load_yaml_config(args.yaml)
+
+    # Check yaml file content
+    is_valid, msg = validate_cluster_config(user_config)
+    if not is_valid:
+        logger.error(f"Cluster Validation: {msg}")
+        raise ValueError(msg)
+
+    if user_config["cluster_type_config"] == "aws":
+        try:
+            aws_infra = AWSInfrastructure(user_config, logger)
+
+            key_pair_id = aws_infra.import_key_pair()
+            vpc_id = aws_infra.create_vpc()
+            subnet_id = aws_infra.create_subnet(vpc_id)
+            igw_id = aws_infra.setup_internet_gateway(vpc_id)
+            rtb_id = aws_infra.create_route_table(vpc_id, igw_id, subnet_id)
+            sg_id = aws_infra.create_security_group(vpc_id)
+            aws_infra.setup_security_group_rules(sg_id)
+            instance_ids = aws_infra.launch_instances(sg_id, subnet_id)
+
+            cluster_config = aws_infra.create_cluster_config(instance_ids)
+            # TODO(pratik): waits for ports to open, do something better here.
+            time.sleep(60)
+
+        except Exception as e:
+            logger.error(f"Error occurred: {e}. Initiating cleanup.")
+            aws_infra.cleanup_resources()
+            raise
+
+    elif user_config["cluster_type_config"] == "azure":
+        try:
+            azure_infra = AzureInfrastructure(user_config, logger)
+
+            resource_group = azure_infra.create_resource_group()
+            _, subnet = azure_infra.create_vnet_and_subnet()
+            public_ip = azure_infra.create_public_ip()
+            azure_infra.deploy_infrastructure(public_ip)
+            azure_infra.create_and_configure_nsg()
+            cluster_config = azure_infra.generate_config_json()
+            azure_infra.mount_disk(cluster_config)
+            # TODO(pratik): waits for ports to open, do something better here.
+            time.sleep(120)
+        except Exception as e:
+            logger.error(f"An error occurred: {e}")
+            azure_infra.cleanup_resources()
+            raise
+
+    elif user_config["cluster_type_config"] == "self-hosted":
+        cluster_config = {}
+        pass
+
+    else:
+        raise ValueError("Cluster Type Not Supported.")
+
+    user_cluster_config = merge_dictionaries(user_config, cluster_config)
+
+    try:
+        validator = ClusterValidator(user_cluster_config, logger)
+        result = validator.validate_cluster()
```

Comment:
This is a good question! I thought to just log the error rather than explicitly stopping execution because there could be cases where you may not need to suffice all the conditions, let's say for cases where you are deploying with a nfs, you may not need all the ports like 2049 to be open. Also, some of these validation commands are bit system dependent, and could face issues on some enterprise clusters. It may just be better once someone faces an error, they can see the logs, and solve it and rerun it. What do you think about this?